### DISCUSSION TITLE
display setcards1.gif closes window after 3 seconds

### DISCUSSION THREAD

**Discussion by BenCollver** [ImageMagick/ImageMagick#2573] [Help] [answered]
When i run "display setcards1.gif" the window closes 3 seconds after the image is displayed. This is not an animated GIF. The file is linked below [1].

I can work around this problem by running "display -loop 9999 setcards1.gif" or by converting the image to another format.

Below [2] is information from my system. Any ideas what is going on here?

Thanks!

-Ben

[1]
https://bencollver.bitbucket.io/setjs/setcards1.gif

[2]
$ uname -a
Linux samtosha 5.8.4-200.fc32.x86_64 #1 SMP Wed Aug 26 22:28:08 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

$ cat /etc/redhat-release
Fedora release 32 (Thirty Two)

$ display -version
Version: ImageMagick 6.9.11-27 Q16 x86_64 2020-08-11 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP(4.5)
Delegates (built-in): bzlib cairo djvu fftw fontconfig freetype gslib gvc jbig jng jp2 jpeg lcms lqr ltdl lzma openexr pangocairo png ps raqm raw rsvg tiff webp wmf x xml zlib

**Comment by urban-warrior** [comment, marked as answer]
Your image likely has a delay in it and the `display` utility honors it.  Try adding `-delay 0` to your command line.  A delay of zero means to persist the image.

**Comment by BenCollver** [comment]
Thanks!  Adding -delay 0 to my command line works around the issue.  This image does indeed have a delay in it.

$ identify -verbose setcards1.gif | grep Delay:
  Delay: 2x100
